packages/mdc-textfield/index.js

```diff
@@ -206,14 +209,103 @@ class MDCTextField extends MDCComponent {
    * @return {boolean} True if the Text Field is required.
    */
   get required() {
-    return this.foundation_.isRequired();
+    return this.input_.required;
   }
 
   /**
    * @param {boolean} required Sets the Text Field to required.
    */
   set required(required) {
-    this.foundation_.setRequired(required);
+    this.input_.required = required;
+  }
+
+  /**
+   * @return {string} the input element's validation pattern.
+   */
+  get pattern() {
+    return this.input_.pattern;
+  }
+
+  /**
+   * @param {string} pattern Sets the input element's validation pattern.
+   */
+  set pattern(pattern) {
+    this.input_.pattern = pattern;
+  }
+
+  /**
+   * @return {number} the input element's minLength.
+   */
+  get minLength() {
+    return this.input_.minLength;
+  }
+
+  /**
+   * @param {number} minLength Sets the input element's minLength.
+   */
+  set minLength(minLength) {
+    this.input_.minLength = minLength;
+  }
+
+  /**
+   * @return {number} the input element's maxLength.
+   */
+  get maxLength() {
+    return this.input_.maxLength;
+  }
+
+  /**
+   * @param {number} maxLength Sets the input element's maxLength.
+   */
+  set maxLength(maxLength) {
+    if (maxLength < 0) {
```

Comment:
I assume this normalization is here for IE and isn't present for minLength because it doesn't support that? I guess this is fair although it makes this smarter than simply being a proxy to the input. Add a comment explaining it though.

Reply:
Well if I set maxLength to < 0, Chrome throws a warning. I will add a comment.